lenny/core/api.py

```diff
@@ -53,11 +53,27 @@ def make_url(cls, path):
         return f"{url}{path}"
 
     @classmethod
-    def auth_check(cls, book_id: int, email=None):
-        # TODO: permission/auth checks go here
-        # for userid, store/check hashed email in db?
+    def auth_check(cls, openlibrary_edition: int, email = None , logged_in: bool = False):
+        """
+        Checks if the user is allowed to access the book.
+        """
+        from lenny.core.models import Item
+        item = Item.exists(openlibrary_edition)
+        if not item:
+            raise ItemExistsError(f"Item with OpenLibrary edition {openlibrary_edition} does not exist.")
```

Comment:
Note to @mekarpeles: In the /read endpoint, are we catching this `ItemExistsError`?